### DISCUSSION TITLE
Developer Tools - Alan Pope's blog

### DISCUSSION THREAD

**Discussion by giscus** [popey/popey.com-blog#26] [General]
# Developer Tools - Alan Pope's blog

I have long said I’m not really a developer. Whenever I used to see news articles in the past quoting me as “Alan Pope, Developer at Canonical”, I would cringe quite a bit. I say to my professional developer friends that I’m not one, and they often roll their eyes at me.
What makes someone a developer though?
I have a GitHub account. I have developed code myself. I’ve badly written HTML, BASH shell scripts, JavaScript, Python, and even some (finished, and some unfinished) “games” in BlitzBasic, BlitzMax, Z80 & 6502 assembler.

https://popey.com/blog/2023/09/developer-tools/

**Comment by thelovebug** [comment]
[Sorry, I had to](https://web.archive.org/web/20120704183707/http://shotofjaq.org/2010/04/01/) - NSFW

**Reply by popey** [reply]
"Shot of Jaq"...

![image](https://github.com/popey/popey.com-blog/assets/1841272/a60ae2fa-84fd-4c54-8e8b-b42e60dcf32c)

**Comment by ajfriesen** [comment]
I feel the same.
My current job title is software and systems engineer.
However I am more in the systems engineer part since I have done mostly task related to this at my job. Infra on AWS with k8s and infra as code. Now I am building paas services for a cloud provider, like k8s as a service, database as a service, etc.

Sure I have written small go tools, bash and python scripts.
But somehow I think, a software engineer needs to solve big problems in a huge enterprise domain.
Also some colleague tell me that I am a software engineer, since I think like one.
So maybe I am 😅

**Comment by fimion** [comment]
So, a few things:

1. I've never been to college, but I am a developer.
2. i did theatre for 15 years, and when i described the scheduling database and tooling i set up to track jobs for my place of employment, people at developer meetups would look at me and be like "... you're a developer. start saying it."
3. ... you're a developer. start saying it.

**Comment by Morgenkaff** [comment]
I have always seen the "function"/profession developer less as an specific use of tools or training ground. And far more as the liquid developer used for developing photos.
Its some process/person(s) you get your project through, to get the finished result.
While it might difficult or even impossible to develope software without writing code, I don't see it as necessary.
